### ISSUE TITLE
Shape.bounds returns incorrect rectangle

### ISSUE THREAD

**Issue by MhdHejazi** [exyte/Macaw#623]
Consider the following simple shape:
`<path d="M160 80.3848C160 187.051 0 187.051 0 80.3848C0 -26.2819 160 -26.2819 160 80.3848Z" fill="#C4C4C4"/>`
![image](https://user-images.githubusercontent.com/121827/65593159-99ceca00-df98-11e9-8d6e-e714ad2f37ac.png)

As you can see the size is 160x160, but when I call `Shape.bounds` it returns:
`x: 0.000000, y: -26.281900, w: 160.000000, h: 213.332900`

If we check the curves of this path closely we can understand what's happening. The returned rectangle is bigger because it also includes the control points:
![image](https://user-images.githubusercontent.com/121827/65593910-347bd880-df9a-11e9-93fc-9388f4850fae.png)

I checked how the bounds are calculated and found out that the native method `CGContext. boundingBoxOfPath` is the culprit since it calculates the bounding box for all the points including the control points:

> ### [CGContext.boundingBoxOfPath](https://developer.apple.com/documentation/coregraphics/cgcontext/1454577-boundingboxofpath)
> The bounding box is the smallest rectangle completely enclosing all points in a path, **_including control points_** for Bézier cubic and quadratic curves.

While `CGPath.boundingBoxOfPath`, despite having the exact same name, doesn't include the control points in the calculation:

> ### [CGPath.boundingBoxOfPath](https://developer.apple.com/documentation/coregraphics/cgpath/1411200-boundingboxofpath)
> The path bounding box is the smallest rectangle completely enclosing all points in the path **_but not including control points_** for Bézier and quadratic curves.

I tried using `ctx.path?.boundingBoxOfPath` instead of `ctx.boundingBoxOfPath` here:
https://github.com/exyte/Macaw/blob/3acbaffa97f0297cde7eb4fce40cdc2e4ca0f56f/Source/model/scene/Shape.swift#L68
and it returned the correct result:
`x: 0.000000, y: 0.384775, w: 160.000000, h: 159.999675`

Searching for `boundingBoxOfPath` in this repo I found out that `Path.bounds()` actually uses the correct method:
https://github.com/exyte/Macaw/blob/3acbaffa97f0297cde7eb4fce40cdc2e4ca0f56f/Source/model/geom2d/Path.swift#L15-L17

So I think this method can be used as a workaround until this is fixed (instead of `Shape.bounds`, one can use `Shape.form.bounds()`).

**Comment by ystrot** [comment]
Hi @MhdHejazi,

Thanks for detailed description! I've pushed a fix.

**Comment by ystrot** [comment]
Actually it's not that easy. We have simple sample in tests:
```xml
<path d="M10,10C40,10,65,10,95,80S150,150,180,80"/>
```

And on this sample both `CGPath.boundingBoxOfPath` and `CGContext. boundingBoxOfPath` works incorrectly (blue is path bounds, red is context bounds):

<img width="195" alt="Screenshot 2019-11-29 at 17 58 56" src="https://user-images.githubusercontent.com/477610/69864865-72ff7d80-12d2-11ea-9c95-ece37ea7a289.png">

In this case we need to use context, because at least it returns a bounding box. It's not minimal, but path bounding box completely wrong. So I'll revert this change and we need to wait Apple will fix it.

**Comment by MhdHejazi** [comment]
I see.
But it's confusing that `Shape.bounds` doesn't return the same value as `Path.bounds()`. The former returns the bigger rectangle in your example, while the later returns the smaller one.

**Comment by ystrot** [comment]
Agree, but it's the best we can do. In `Path.bounds` we don't have context, so we need to use path method.
